src/features/app-interfaces/pages/edit-app-interface-page.tsx

```diff
@@ -12,7 +12,15 @@ import { useTitle } from '@/utils/use-title'
 
 const transformError = (e: Error) => {
   if (is404(e)) {
-    return new Error(appInterfaceNotFoundMessage)
+    const error = new Error(appInterfaceNotFoundMessage) as StatusError
+    error.status = 404
+    return error
+  }
+  // This is needed because the App interface not found doesn't return a 404 status code
+  if (e.message.includes('App interface not found')) {
```

Comment:
I am not sure what you mean here? I wasn't seeing that the 404 was being triggered when the app interface was not found. What would we be extracting? The reason for that text is that that was being return by the server in the error
<img width="416" alt="image" src="https://github.com/user-attachments/assets/186009a8-2779-4a3a-97ab-6e08ae846232">

Reply:
I noticed we have 2 different messages for the App inteface not found state. "App interface not found" and "Application Interface not found". We should consolidate this into one. Additionally we should extract this into a constant and use that const when doing the conditional and where the error is throw to ensure that if the message changes the conditional also changes.

Reply:
I see what you are saying. I made the change. The only thing is that I had to add the 404 status in the error. Since the useAppInterface call doesn't return a status code, I added one  in order to show the error with the return link

Reply:
I also had to extract the errors message and labels into another file to avoid circular deps. Well, I only had to extract that one const, but I moved all the labels and message in that file to another file.